src/wp-includes/l10n.php

```diff
@@ -1988,3 +1988,38 @@ function wp_get_word_count_type() {
 
 	return $wp_locale->get_word_count_type();
 }
+
+/**
+ * Parses the `Accept-Language` header to get a list of locales.
+ *
+ * The locales are in the format WordPress expects, so "fr-CH" becomes "fr_CH"
+ * and "fr" becomes "fr_FR". "en" and wildcard values are discarded.
+ * The priority weighting is also discarded.
+ *
+ * @since 6.6.0
+ *
+ * @return string[] Locales list.
+ */
+function get_locales_from_accept_language_header() {
+	$locales = array();
+
+	if ( ! empty( $_SERVER['HTTP_ACCEPT_LANGUAGE'] ) ) {
```

Comment:
We should only pass strings to the `preg_match_all()` call to avoid notices/errors with any other type.